docs/interfaces/README.md

```diff
@@ -1,10 +1,10 @@
 <!--
 order: false
 parent:
-  order: 6
+  order: 7
 -->
 
-# Interfaces
+# Interfaces (Deprecated)
```

Comment:
Why marking this as deprecated as opposed to just removing this section entirely? 

It seems like a number of these subsections are pretty much out of date and/or made redundant by the new docs proposed in this PR. Specifically:
- "introduction to interfaces" can probably go
- "rest inteerfaces" can probably go
- "CLI" & "Lifecycle of a Query" sections maybe should rather be updated? Not totally sure..

Either way, i'm in favor of merging this PR as is as long as we have explicit issues / notes in existing issues to clean up the rest of these. Thoughts?

Reply:
yeah exactly, i added an item in https://github.com/cosmos/cosmos-sdk/issues/7661 to do this exactly.